src/main/java/org/dita/dost/module/reader/AbstractReaderModule.java

```diff
@@ -639,84 +640,91 @@ void outputResult() throws DITAOTException {
         resourceOnlySet.addAll(res);
 
         for (final URI file: outDitaFilesSet) {
-            getOrCreateFileInfo(fileinfos, file).isOutDita = true;
+            createOrUpdateFileInfo(file, fi -> fi.isOutDita = true);
```

Comment:
Instead of getting the created or old `FileInfo` and updating it directory, we pass an update function that does the updating.